.codecov.yml

```
@@ -0,0 +1,49 @@
+---
+
+codecov:
+  notify:
+    after_n_builds: 18  # Number of test matrix+lint jobs uploading coverage
+    wait_for_ci: false
+
+  require_ci_to_pass: false
+
+  token: >-  # notsecret  # repo-scoped, upload-only, stability in fork PRs
+    xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

@samdoran edit this comment with the actual *repo*-scoped token and commit it:
```suggestion
    xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

Repo-scoped tokens aren't as secret. It's now an established practice to save them as clear text. It improves stability of the uploads and is available to PRs from forks. There's also a new setting on Codecov UI that allows disabling the requirement to use tokens at all, if you'd want to go that way, but I haven't yet tested it. Plus, it probably needs a newer version of the Codecov action.

Note that w/o this, coverage is not being uploaded from `main` at all it fails silently and the CI remain green: https://github.com/samdoran/ansible-collection-macos/actions/runs/11352888972/job/31576657080#step:3:1110.

---

Got it. I wanted to read through all the docs and look at examples so I understood this fully. Thanks for adding this.